### PR TITLE
Patch flaky zkpersister test

### PR DESCRIPTION
**Pull request by szabado**
Patch this so we stop having flaky pipelines.

### CODE REVIEW

**Comment by szabado** [comment]
This depends on https://github.com/square/quotaservice/pull/140, that should be merged first

**Comment by szabado** [comment]
The test here wasn't actually flaky, it was the underlying code.

The patch was moving the watcher re-establishment before we get the list of current configs.  Worst case scenario, this _can_ result in an extra call to `currentConfigEventListener` but it fixes some of the raciness and makes the test not flake out on us.  The extraneous call to `currentConfigEventListener` shouldn't have any effect, as it'll just be fetching configs it already knows exist.

File: config/zookeeper_persister.go [hunk 1]
```diff
@@ -208,11 +208,31 @@ func (z *ZkConfigPersister) ReadPersistedConfig() (*pb.ServiceConfig, error) {
...
+	// Placing this before the z.conn.Children() ensures we won't miss any new versions that are created during the
+	// search
+	_, _, ch, err := z.conn.GetW(z.path)
+
+	if err != nil {
+		logging.Printf("Received error from zookeeper when fetching %s: %+v", z.path, err)
+		return nil, err
+	}
+
 	if z.initialized {
```
Comment (spenceral) [inline config/zookeeper_persister.go]: Will the value of `z.initialized` ever have changed from when it is checked above?

Comment (szabado) [reply config/zookeeper_persister.go]: I don't believe so

**Pull request merged**